example/poisson_gmg/poisson_package.cpp

```diff
@@ -77,60 +80,60 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   Real err_tol = pin->GetOrAddReal("poisson", "error_tolerance", 1.e-8);
   pkg->AddParam<>("error_tolerance", err_tol);
 
+  Real res_tol = pin->GetOrAddReal("poisson", "residual_tolerance", 1.e-8);
+  pkg->AddParam<>("residual_tolerance", res_tol);
+
   bool fail_flag = pin->GetOrAddBoolean("poisson", "fail_without_convergence", false);
   pkg->AddParam<>("fail_without_convergence", fail_flag);
 
   bool warn_flag = pin->GetOrAddBoolean("poisson", "warn_without_convergence", true);
   pkg->AddParam<>("warn_without_convergence", warn_flag);
 
-  // res_err enters a multigrid level as the residual from the previous level, which
-  // is the rhs, and leaves as the solution for that level, which is the error for the
-  // next finer level
-  auto te_type = Metadata::Cell;
-  if (GetTopologicalType(te) == parthenon::TopologicalType::Node) {
-    te_type = Metadata::Node;
-  } else if (GetTopologicalType(te) == parthenon::TopologicalType::Edge) {
-    te_type = Metadata::Edge;
-  } else if (GetTopologicalType(te) == parthenon::TopologicalType::Face) {
-    te_type = Metadata::Face;
-  }
-  using namespace parthenon::refinement_ops;
-  auto mres_err = Metadata({te_type, Metadata::Independent, Metadata::FillGhost,
-                            Metadata::GMGRestrict, Metadata::GMGProlongate});
-  mres_err.RegisterRefinementOps<ProlongateSharedLinear, RestrictAverage>();
-  pkg->AddField(res_err::name(), mres_err);
+  bool use_exact_rhs = pin->GetOrAddBoolean("poisson", "use_exact_rhs", false);
+  pkg->AddParam<>("use_exact_rhs", use_exact_rhs);
+
+  PoissonEquation eq;
+  eq.do_flux_cor = flux_correct;
+
+  parthenon::solvers::MGParams mg_params;
+  mg_params.max_iters = max_poisson_iterations;
+  mg_params.residual_tolerance = res_tol;
+  mg_params.do_FAS = do_FAS;
+  mg_params.smoother = smoother_method;
+  parthenon::solvers::MGSolver<u, rhs, PoissonEquation> mg_solver(pkg.get(), mg_params,
+                                                                  eq);
+  pkg->AddParam<>("MGsolver", mg_solver, parthenon::Params::Mutability::Mutable);
```

Comment:
This is super nice now... modulo of vourse that one has to implement the `PoissonEquation`